### ISSUE TITLE
getElementsByTagName('tag') returning DOMText

### ISSUE THREAD

**Issue by andrerom** [facebook/hhvm#5134]
This might be the culprit for issues reported in #2843 and elsewhere with the resulting error being:
``Fatal error: Call to undefined method DOMText::hasAttribute()`` / ``DOMText::setAttribute()``

However  DOMText does not have these method, it is rather that on hhvm we seem to get DOMText when we expect only DOMElement.

Example:
```php
        foreach ( $document->getElementsByTagName( 'paragraph' ) as $paragraph )
        {
            if ( $paragraph->hasAttribute( 'xmlns:tmp' ) )
```
Links for this example: [Travis](https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/57035489), [Code](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/FieldType/XmlText/Converter/Expanding.php#L58), [TestCase](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/FieldType/Tests/XmlText/Converter/ExpandingTest.php#L19), & [Fixtures](https://github.com/ezsystems/ezpublish-kernel/tree/master/eZ/Publish/Core/FieldType/Tests/XmlText/Converter/_fixtures/expanding/input)
_Note: Travis is currently on HHVM 3.5, however I could not see any issues relevant to this issue being fixed in 3.6_

**Comment by paulbiss** [comment]
Any chance you could produce a short self-contained example demonstrating the problem? It also may be worth trying this against master as a number of bugs in ext_dom and ext_simplexml were recently fixed by 779b97a7c0a0f1c257f09727b20d86eb0f708e14.

**Comment by andrerom** [comment]
switched to hhvm-nightly and seems to indeed be fixed now, closing.
ref: https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/57335061
